### ISSUE TITLE
Add easy way to "get" a layer/interface/charm

### ISSUE THREAD

**Issue by marcoceppi** [juju/charm-tools#121]
I'm proposing a new set of commands:

`charm get-source`

or

`charm pull-source`

which takes a parameter of:

charm:
 - `[cs:]charm`
 - `[cs:]series/charm`
 - `[cs:]~user/series/charm`
layer:
 - `layer:layer-name`
 - `interface:layer-name`

**Comment by marcoceppi** [comment]
This is a proposed workaround for #121

**Comment by johnsca** [comment]
If the upload and download commands are `charm push` and `charm pull`, then I'm +1 for `charm pull-source`.

The source repo for layers is easy to discover using the API, but most charm layers aren't registered there and we don't currently track the upstream repo for most charm layers.  Thus, I'd also propose we add a (semi-)recommended `repo` field to `layer.yaml`.  When doing a `charm build`, it should (at least) warn if that field is missing and offer a suggested value discovered from the VCS info.  I'd recommend preferring an `upstream` remote for git over `origin`, if present, and `:parent` over `:push` for bzr.  It might need to convert `git://` and `lp:` URLs to `https://`.

**Comment by marcoceppi** [comment]
I'll open a separate issue for tracking that, but it's a great point. Though, I'm starting to think that, most layer-built charms won't have the built charm in vcs, so when you charm push and set "homepage" information, it should probably point to the top layer

**Comment by johnsca** [comment]
Agreed, I would expect most `homepage-url` values to point to the charm layer source repo, but I can also see them being set to the project home page or some other URL.  There is also no linting to ensure that the `homepage-url` is set.

**Comment by marcoceppi** [comment]
yes, I just realized that. Having it in the layer would probably be the most straightforward way
